File: dec11/one.mjs
import fs from "fs/promises";

(async () => {
    const file = (await fs.readFile("input.txt", "utf8")).trim();
    let nums = file.split(' ').map(Number)
    for(let _ = 25; _--;){
        let arr = [];
        for(let i = 0; i < nums.length; i++){
            let num = nums[i];
            let str = num.toString();
            if(num === 0){
                arr.push(1);
            }
            else if(str.length % 2 === 0){
                arr.push(+str.slice(0, str.length / 2));
                arr.push(+str.slice(str.length / 2));
            }
            else {
                arr.push(num * 2024);
            }
        }
        nums = arr;
    }
    console.log(nums.length);
})()

File: dec11/one.rs
use std::fs;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let file = fs::read_to_string("input.txt")?;
    let mut nums: Vec<i64> = file
        .trim()
        .split_whitespace()
        .map(|s| s.parse::<i64>().unwrap())
        .collect();

    for _ in 0..25 {
        let mut arr = Vec::new();
        for &num in &nums {
            let s = num.to_string();
            if num == 0 {
                arr.push(1);
            } else if s.len() % 2 == 0 {
                let mid = s.len() / 2;
                let first_half = &s[..mid];
                let second_half = &s[mid..];
                arr.push(first_half.parse::<i64>().unwrap());
                arr.push(second_half.parse::<i64>().unwrap());
            } else {
                arr.push(num * 2024);
            }
        }
        nums = arr;
    }
    println!("{}", nums.len());
    Ok(())
}